test/src/userobjects/TestShapeElementUserObject.C

```diff
@@ -74,12 +76,21 @@ TestShapeElementUserObject::executeJacobian(unsigned int jvar)
 void
 TestShapeElementUserObject::finalize()
 {
-  // check in all MPI processes if executeJacobian was called for each variable
+  // Using this semi-encapsulated member feels like cheating but hey,
+  // it's a test object.
+  const FEProblemBase & prob = _ti_feproblem;
+
+  const dof_id_type n_local_dfs = prob.getNonlinearSystemBase().system().n_local_dofs();
+
+  // check if executeJacobian was called for each variable on at least
+  // one MPI process.  We'd like to make sure that executeJacobian was
+  // called on *every* process which owns degrees of freedom, but I
+  // don't think we
```

Comment:
Ha!  I was starting to write an explanation of why I couldn't make this test as robust as it should be, then I suddenly had an idea about how to make this test as robust as it should be, then I figured out how to actually implement that idea, did that, tested it... and by this time had completely forgotten that I'd started writing a comment in there too.

It's fixup'ed now.